pkg/cloud/azureclient.go

```diff
@@ -152,18 +152,12 @@ func getClient(env azure.Environment, subscriptionID string, credential azcore.T
 		return nil, errors.Wrap(err, "failed to create request adapter")
 	}
 
-	clientOpts := &armpolicy.ClientOptions{
-		ClientOptions: azcore.ClientOptions{
-			Transport: client,
-		},
-	}
```

Comment:
This http client is specific to the graph sdk and causes issues when used with the `azure-sdk-for-go` clients, so removing it (this was not wired before either)

Reply:
Curious, what issues does it cause?  This means we don't get logs for any calls this code makes right?

Reply:
this is the error

```bash
PUT https://management.azure.com/subscriptions/<SUBSCRIPTION_ID>/resourceGroups/aramasedev/providers/Microsoft.KeyVault/vaults/<KEYVAULT_NAME>/providers/Microsoft.Authorization/roleAssignments/a91b41b8-f7ed-4eaa-8b57-34df7ccb90e4
--------------------------------------------------------------------------------
RESPONSE 400: 400 Bad Request
ERROR CODE: InvalidRequestContent
--------------------------------------------------------------------------------
{
  "error": {
    "code": "InvalidRequestContent",
    "message": "The content of your request was not valid, and the original object could not be deserialized. Exception message: 'Unable to translate bytes [8B] at index 1 from specified code page to Unicode.'"
  }
}
--------------------------------------------------------------------------------
```

Reply:
I have no idea how using the wrapped client would cause that 🤔